### PR TITLE
GH-8070

### PR DESCRIPTION
**Pull request by plabadie**
closes GH-8070

### CODE REVIEW

File: projects/storefrontlib/src/cms-structure/routing/default-routing-config.ts [hunk 1]
```diff
@@ -219,6 +219,9 @@ export const defaultStorefrontRoutesConfig: RoutesConfig = {
   userGroupAssignUsers: {
     paths: ['organization/user-group/assign-users/:code'],
   },
+  orderApprovals: {
+    paths: ['my-account/approval-dashboard'],
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-structure/routing/default-routing-config.ts]: isn't this for the listing page?

File: projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail.service.spec.ts [hunk 2]
```diff
@@ -62,4 +62,11 @@ describe('OrderApprovalDetailService', () => {
     expect(order).toBeTruthy();
     expect(order).toEqual(mockOrder);
   });
+
+  it('should provide the order approval data', () => {
+    let orderApproval: OrderApproval;
+    service.getOrderApproval().subscribe((value) => (orderApproval = value));
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail.service.spec.ts]: maybe we can add a .unsubscribe here after the subscription

File: projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail.service.ts [hunk 3]
```diff
@@ -18,24 +23,29 @@ export class OrderApprovalDetailService {
...
       this.orderApprovalService.loadOrderApproval(approvalCode)
     ),
     switchMap((approvalCode: string) =>
       this.orderApprovalService.get(approvalCode)
     ),
-    pluck('order'),
     shareReplay({ bufferSize: 1, refCount: true })
   );
 
+  protected order$ = this.orderApproval$.pipe(pluck('order'));
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail.service.ts:37]: 👍

File: projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import { Observable } from 'rxjs';
+import { OrderApprovalDetailService } from '../order-approval-detail.service';
+
+@Component({
+  selector: 'cx-order-approval-detail-form',
+  templateUrl: './order-approval-detail-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class OrderApprovalDetailFormComponent {
+  approvalDecision: 'APPROVE' | 'REJECT';
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.ts:13]: to discuss (as mentioned in our previous talk)

File: projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    this.approvalDecision = decision;
+    if (decision === 'APPROVE') {
+      this.approvalForm.controls.comment.clearValidators();
+    } else {
+      this.approvalForm.controls.comment.setValidators([Validators.required]);
+    }
+    this.approvalFormVisible = true;
+  }
+
+  cancelDecisionForm() {
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.ts:39]: wondering if there's an added benefit if we were to add the lifecycle hook for init to call this function

File: projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+    fixture = TestBed.createComponent(OrderApprovalDetailFormComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+    orderApprovalDetailService = TestBed.inject(OrderApprovalDetailService);
+    orderApprovalService = TestBed.inject(OrderApprovalService);
+    el = fixture.debugElement;
+  });
+
+  it('should create', () => {
+    console.log('temp', orderApprovalDetailService, orderApprovalService);
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts]: need to remove

File: projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+    displayDecisionForm(APPROVE);
+    expect(component.approvalForm.valid).toBeTrue();
+  });
+
+  it('should comment be required for rejection.', () => {
+    displayDecisionForm(REJECT);
+    expect(component.approvalForm.valid).toBeFalse();
+  });
+
+  it('should submit rejection form.', () => {});
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts]: duplicate from above but empty

File: projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+
+  it('should display approval form when approve order button clicked, and then hide form on cancel.', () => {
+    displayAndCancelDecisionForm(APPROVE);
+  });
+
+  it('should display rejection form when reject order button clicked, and then hide form on cancel.', () => {
+    displayAndCancelDecisionForm(REJECT);
+  });
+
+  it('should submit approval form.', () => {
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts]: should submit approval form with comments, same as the rejected one below

File: projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+    displayDecisionForm(APPROVE);
+    submitDecisionForm(APPROVE);
+  });
+
+  it('should submit rejection form.', () => {
+    displayDecisionForm(REJECT);
+    submitDecisionForm(REJECT);
+  });
+
+  it('should comment be optional for approval.', () => {
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts]: I think it's best to say should have comment as optional for approval

File: projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+    displayDecisionForm(APPROVE);
+    expect(component.approvalForm.valid).toBeTrue();
+  });
+
+  it('should comment be required for rejection.', () => {
```
Comment (bgambocjaviniar) [inline projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.spec.ts]: same as above, but maybe we can add an added assertion where we click it and it validation is still false

File: projects/storefrontlib/src/cms-structure/routing/default-routing-config.ts (hunk 1 above)
Comment (plabadie) [reply projects/storefrontlib/src/cms-structure/routing/default-routing-config.ts]: Exact, I added this for the "Back to List" button to work.

Comment (bgambocjaviniar) [reply projects/storefrontlib/src/cms-structure/routing/default-routing-config.ts]: got it

File: projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.ts (hunk 4 above)
Comment (bgambocjaviniar) [reply projects/storefrontlib/src/cms-components/organization/order-approval/order-approval-detail/order-approval-detail-form/order-approval-detail-form.component.ts:13]: refer to https://github.com/SAP/spartacus/issues/8145

**Review by bgambocjaviniar — APPROVED** [review]

**Pull request merged**